File: frontend/frontend/src/app/core/brand-colors.util.ts
const DEFAULT_PRIMARY = '#0ea5e9';
const DEFAULT_PRIMARY_DEEP = '#0284c7';

export function normalizeHex(hex: string | null | undefined): string {
  if (!hex?.trim()) return DEFAULT_PRIMARY;
  let h = hex.trim();
  if (!h.startsWith('#')) h = `#${h}`;
  if (h.length !== 7 && h.length !== 4) return DEFAULT_PRIMARY;
  return h.toLowerCase();
}

/** Escurece um hex #RRGGBB (factor 0–1). */
export function darkenHex(hex: string, factor: number): string {
  const n = normalizeHex(hex);
  if (n.length === 4) return DEFAULT_PRIMARY_DEEP;
  const r = parseInt(n.substring(1, 3), 16);
  const g = parseInt(n.substring(3, 5), 16);
  const b = parseInt(n.substring(5, 7), 16);
  const f = Math.max(0, Math.min(1, factor));
  const dr = Math.round(r * (1 - f));
  const dg = Math.round(g * (1 - f));
  const db = Math.round(b * (1 - f));
  return `#${dr.toString(16).padStart(2, '0')}${dg.toString(16).padStart(2, '0')}${db
    .toString(16)
    .padStart(2, '0')}`;
}

export function brandPalette(primary?: string | null): { primary: string; primaryDeep: string } {
  const p = normalizeHex(primary);
  const d = darkenHex(p, 0.22);
  return { primary: p, primaryDeep: d };
}

/** Substitui cores padrão Aero Suite em HTML inline de impressão/e-mail. */
export function applyBrandPalette(
  html: string,
  primary?: string | null,
  primaryDeep?: string | null
): string {
  if (!html) return html;
  const p = normalizeHex(primary);
  const d = normalizeHex(primaryDeep ?? darkenHex(p, 0.22));
  if (p === DEFAULT_PRIMARY && d === DEFAULT_PRIMARY_DEEP) return html;
  return html
    .replaceAll(DEFAULT_PRIMARY, p)
    .replaceAll(DEFAULT_PRIMARY.toUpperCase(), p)
    .replaceAll(DEFAULT_PRIMARY_DEEP, d)
    .replaceAll(DEFAULT_PRIMARY_DEEP.toUpperCase(), d);
}

File: backend/src/main/java/com/aerosuite/util/BrandColorsUtil.java
package com.aerosuite.util;

/**
 * Paleta de marca para HTML/PDF (substitui cores padrão Aero Suite).
 */
public final class BrandColorsUtil {

    public static final String DEFAULT_PRIMARY = "#0ea5e9";
    public static final String DEFAULT_PRIMARY_DEEP = "#0284c7";

    private BrandColorsUtil() {}

    public static String normalizeHex(String hex) {
        if (hex == null || hex.isBlank()) {
            return DEFAULT_PRIMARY;
        }
        String h = hex.trim();
        if (!h.startsWith("#")) {
            h = "#" + h;
        }
        if (h.length() != 7 && h.length() != 4) {
            return DEFAULT_PRIMARY;
        }
        return h.toLowerCase(java.util.Locale.ROOT);
    }

    /** Escurece um hex #RRGGBB (factor 0–1). */
    public static String darken(String hex, double factor) {
        String n = normalizeHex(hex);
        if (n.length() == 4) {
            return DEFAULT_PRIMARY_DEEP;
        }
        int r = Integer.parseInt(n.substring(1, 3), 16);
        int g = Integer.parseInt(n.substring(3, 5), 16);
        int b = Integer.parseInt(n.substring(5, 7), 16);
        double f = Math.max(0, Math.min(1, factor));
        r = (int) Math.round(r * (1 - f));
        g = (int) Math.round(g * (1 - f));
        b = (int) Math.round(b * (1 - f));
        return String.format(java.util.Locale.ROOT, "#%02x%02x%02x", r, g, b);
    }

    public static String applyBrandPalette(String html, String primary, String primaryDeep) {
        if (html == null || html.isEmpty()) {
            return html;
        }
        String p = normalizeHex(primary);
        String d = normalizeHex(primaryDeep != null && !primaryDeep.isBlank() ? primaryDeep : darken(p, 0.22));
        if (DEFAULT_PRIMARY.equalsIgnoreCase(p) && DEFAULT_PRIMARY_DEEP.equalsIgnoreCase(d)) {
            return html;
        }
        return html
                .replace(DEFAULT_PRIMARY, p)
                .replace(DEFAULT_PRIMARY.toUpperCase(java.util.Locale.ROOT), p)
                .replace(DEFAULT_PRIMARY_DEEP, d)
                .replace(DEFAULT_PRIMARY_DEEP.toUpperCase(java.util.Locale.ROOT), d);
    }
}
